### ISSUE TITLE
ObjectDetection2DSet Comment May be Incorrect

### ISSUE THREAD

**Issue by derkmed** [PTG-Kitware/angel_system#324]
https://github.com/PTG-Kitware/angel_system/blob/49c93918636eca96ceaf188bfb2b7235b8701bbc/ros/angel_msgs/msg/ObjectDetection2dSet.msg#L45

When subscribing to topics that  publish this message, I did not read a flattened matrix. Instead I read a vector of confidence scores corresponding to the label_vec.

**Comment by Purg** [comment]
This comment is "correct", but a previous incarnation of the object detector was populating the message incorrectly. The newest Yolo detector that we have added should now be populating this field correctly.

**Comment by Purg** [comment]
@derkmed Checking in if this explanation was sufficient to close this issue or if there is still something to be resolved.

**Comment by derkmed** [comment]
Got it. I will double check @Purg . So, each row should correspond to a detected object indicated via one-hot vector. Is that correct?

**Comment by Purg** [comment]
> Got it. I will double check @Purg . So, each row should correspond to a detected object indicated via one-hot vector. Is that correct?

Yes, but the vector is not specifically constrained to be one-hot: it is a per-class confidence vector. Some detector implementations may *choose* to populate this vector in a one-hot fashion.
